source/base/quadrature_lib.cc

```diff
@@ -1777,6 +1778,26 @@ QGaussSimplex<dim>::QGaussSimplex(const unsigned int n_points_1D)
            Utilities::to_string(n_points_1D)));
 }
 
+template <int dim>
+QSupportPointsSimplex<dim>::QSupportPointsSimplex(const unsigned int fe_degree)
+  : QSimplex<dim>(Quadrature<dim>())
+{
+  // fill quadrature points
+  if (dim == 1)
+    {
+      const dealii::QGaussLobatto<dim> quad(fe_degree + 1);
+      this->quadrature_points = quad.get_points();
+    }
+  else
+    {
+      FE_SimplexP<dim> fe(fe_degree);
+      this->quadrature_points = fe.get_unit_support_points();
```

Comment:
I would rather use this directly:
```cpp
Quadrature<dim> quadrature(fe.get_unit_support_points());
```
This effect should be the same without the need to create a new class.

Reply:
Unfortunately not, within the `matrix_free `object `MappingInfo` is trying to create a face quadrature, which fails for Qudarature<dim>(dof_handler_u.get_fe().get_unit_support_points()) as the quadrature doesn't know any rule how to create the face quadrature.

Reply:
I would nevertheless not create a new publicly available (dummy) class. One can keep everything local to `include/deal.II/matrix_free/util.h`. Either one moves `QSupportPointsSimplex` there into an internal namespace or one calls `FE_SimplexP::get_unit_support_points()` or `internal::unit_support_points_fe_p()`: https://github.com/dealii/dealii/blob/94532f1e38c47a21adddaeb00d09a12dab284fe2/source/fe/fe_simplex_p.cc#L117

 I would prefer latter.

Reply:
I agree; the overall design is a bit of a workaround in the sense that we try to figure out the quadrature rule for the face, given what kinds of points we detect for the cell (which is a consequence of the fact that we had tensor products initially where one works with 1D quadrature formulas). But we need to make the best out of the situation now, so it seems like a pragmatic way forward to
- detect whether we have a quadrature formula on the support points of the cell,
- then generate, with the `unit_face_support_points` function, the associated quadrature formula for the faces necessary for the matrix-free object in deal.II.

Reply:
Ok, it now works like you two suggested, first it is checked if the quadrature is on the support points, if yes the face support points are inserted.